cinema/views.py

```diff
@@ -0,0 +1,47 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+class MovieCRUDView(APIView):
+    model = Movie
+    serializer_class = MovieSerializer
+
+    def get_object(self, pk: int):
+        instance = get_object_or_404(self.model, pk=pk)
+        return instance
+
+    def get(self, request, *args, **kwargs):
+        pk = kwargs.get("pk")
+        if pk is not None:
+            movie = self.get_object(pk)
+            serializer = self.serializer_class(movie)
+        else:
+            movies = self.model.objects.all()
+            serializer = self.serializer_class(movies, many=True)
+        return Response(serializer.data, status.HTTP_200_OK)
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=request.data)
+        if serializer.is_valid(raise_exception=True):

```

Comment:
There is no need for the `if` condition, remove it and leave the `is_valid` call with `raise_exception=True` parameter. The code will not execute further in case an exception is raised. Fix in all places.

Reply:
not fixed